src/Microsoft.Framework.Configuration/ConfigurationSection.cs

```diff
@@ -65,14 +87,14 @@ public string Value
 
                 foreach (var src in Sources)
                 {
-                    src.Set(Key, value);
+                    src.Set(Path, value);
                 }
             }
         }
 
         protected override string GetPrefix()
         {
-            return _key + Constants.KeyDelimiter;
+            return Path + Constants.KeyDelimiter;
```

Comment:
Actually maybe we don't need it anymore if we change the argument of the constructor to be `parentPath` instead of prefix and we concatenate the key delimiter there. Then we can just use the `Path` property. It sounds like it would clean up things.